src/restore.ts

```diff
@@ -54,7 +54,25 @@ async function run(): Promise<void> {
             const isExactKeyMatch = utils.isExactKeyMatch(primaryKey, cacheKey);
             utils.setCacheHitOutput(isExactKeyMatch);
 
-            core.info(`Cache restored from key: ${cacheKey}`);
+            let foundKey: string;
+            if (isExactKeyMatch) {
+                foundKey = primaryKey;
+                core.info(`Cache restored for key: ${foundKey}`);
+            } else {
+                let i: number;
+                for (i = 0; i < restoreKeys.length - 1; i++) {
+                    const fallbackCacheKey = await cache.restoreCache(
```

Comment:
Why are you calling `restoreCache` again?
See line 36: it is already called with the restore keys, so how can this succeed?